### DISCUSSION TITLE
Have different actions based on button ? / How to get formEvent handler's target ?

### DISCUSSION THREAD

**Discussion by VaZark** [mantinedev/mantine#5209] [Q&A] [answered]
In vanilla HTML, we have the option to associate submit inputs/buttons with a _formaction_ attribute that triggers a different URL  instead of the default action set up in the form tag. 

In pure React, we work around this by detecting the name associated with the event target. 

```js
 onSubmit = (e) => {
    e.preventDefault();   
    const formAction = e.target.name
    if (formaction === 'download') {
    } else if (formaction === 'stop') {
    }
}
```

However by overriding the submit with only the type-safe input values like `onSubmit={form.onSubmit(handleSearchSubmit)}`, we're missing the event target information that allows us to use this pattern.

Is there a way we can use this pattern with Mantine's useForm? If not, how can we trigger different actions for the same form with different buttons?

**Comment by rtivital** [comment, marked as answer]
It is not required to use `onSubmit={form.onSubmit(handleSearchSubmit)}`. You can replace it with your own submit function.

Example:

```tsx
<form onSubmit={event => {
  event.preventDefault();
  if (form.validate().hasErrors) {
    // handle errors here
  } else {
    // logic without errors here
  }
}} />
```

**Reply by VaZark** [reply]
Thanks!
